rpcperms/interceptor.go

```diff
@@ -751,6 +778,22 @@ func (r *InterceptorChain) checkRPCState(srv interface{}) error {
 
 		return ErrRPCStarting
 
+	// If lnd is waiting for the remote signer to connect, we only allow
+	// calls to the remote signer.
+	case allowRemoteSigner:
+		_, ok := srv.(lnrpc.WalletUnlockerServer)
+		if ok {
+			return ErrWalletUnlocked
```

Comment:
This should be `ErrWalletLocked`, I think.

Reply:
This case only happens if an `lncli unlock` is made on the watch-only node, after the the wallet has already been unlocked. Therefore I do think `ErrWalletUnlocked` is correct here. I did edit the error below though, the previous `ErrRPCStarting` that was currently returned if an RPC request was made, other than the RPC request to connect an outbound remote signer. I edited that to a new custom error that informs the user that they need to connect a remote signer before other calls can be accepted.